main/openrc/modloop.initd

```diff
@@ -99,7 +99,35 @@ start() {
 		mount -t overlay -o upperdir=/.modoverlayfs/modules,lowerdir=/.modloop/modules,workdir=/.modoverlayfs/work overlay /lib/modules
 		eend $? || return 1
 	else
-		rm -rf /lib/modules && ln -sf /.modloop/modules /lib/
+		# if we have a directory /lib/modules/$(uname -r)/local with modules, preserve these
+		LOCAL_MODULE_DIR=/lib/modules/$(uname -r)/local
+		if [ -z $KOPT_local_modules ]; then
+			if [ ! -d $LOCAL_MODULE_DIR ]; then
+				mkdir $LOCAL_MODULE_DIR
+			fi
+			ebegin "Combining local kernel modules with /.modloop/modules"
+			# move local modules dir to tmp
+			mv $LOCAL_MODULE_DIR /tmp
+			# remove all modules and link all directories from /.modloop/modules/$(uname -r)/
+			rm -rf /lib/modules && mkdir -p /lib/modules/$(uname -r)
+			for d in $(find /.modloop/modules/$(uname -r)/ -mindepth 1 -maxdepth 1 -type d)
+			do
+				ln -sf $d /lib/modules/$(uname -r)
+			done
+			# copy original depmod files (to supress depmod warning)
+			cp /.modloop/modules/$(uname -r)/* /lib/modules/$(uname -r)/ 2>/dev/null
+			# link firmware 
+			ln -sf /.modloop/modules/firmware /lib/modules
+			# move local modules back
+			mv /tmp/local /lib/modules/$(uname -r)
```

Comment:
... so this would fail because `/lib/modules` is now a symlink into the read-only `/.modloop`.

Reply:
No, the subdirs of /.moduloop/modules/$(uname -r)/ are linked, not the dir itself. That's why I'm able to create the local dir at the same level. And, it works (I use it).